### ISSUE TITLE
Fatal Exception: mutex lock failed: Invalid argument on iOS

### ISSUE THREAD

**Issue by ynnckcmprnl** [Adobe-Marketing-Cloud/acp-sdks#473]
Hi,

we've been seeing a rather large number of crashes in our iOS app concerning the Adobe SDK. It seems to be a crash buried deep into the SDK, hence I'm posting this here.

It seems to be occurring on all iOS firmwares (12, 13 & 14) and a wide variety of devices, but mostly older devices. Crashlytics is reporting that all crashes happen during background state.

Using these versions of the SDK, we've initially encountered the crash on older SDK versions and have updated to these hoping it would solve the issue. New versions of the SDK have been released, but this is what we currently have in production:

```
PODS:
  - ACPAnalytics (2.2.4):
    - ACPAnalytics/iOS (= 2.2.4)
    - ACPCore (>= 2.5.0)
  - ACPAnalytics/iOS (2.2.4):
    - ACPCore (>= 2.5.0)
  - ACPCore (2.6.2):
    - ACPCore/iOS (= 2.6.2)
  - ACPCore/iOS (2.6.2)
  - ACPTarget (2.1.6):
    - ACPCore (>= 2.5.0)
    - ACPTarget/iOS (= 2.1.6)
  - ACPTarget/iOS (2.1.6):
    - ACPCore (>= 2.5.0)
```

```
Fatal Exception: std::__1::system_error
mutex lock failed: Invalid argument

std::__1::shared_ptr<AdobeMarketingMobile::AlertListenerInterface>::~shared_ptr() + 4349337804
```

Any clues? Could this be resolved in the latest versions?

![Screenshot 2020-11-17 at 13 06 36](https://user-images.githubusercontent.com/690119/99389761-ce59bf80-28d7-11eb-9803-efa07e13ae72.jpg)
![Screenshot 2020-11-17 at 13 07 06](https://user-images.githubusercontent.com/690119/99389768-d154b000-28d7-11eb-93ac-b3aeb0f1e58e.jpg)

**Comment by shalehaha** [comment]
@ynnckcmprnl 
Please upgrade to the latest SDK, we made some fixes on version 2.7.1 https://aep-sdks.gitbook.io/docs/release-notes#july-16-2020.

**Comment by ynnckcmprnl** [comment]
Updating to the latest SDK resolved this issue.
